### PR TITLE
chore: replace usage of MCore::LinearInterpolate with AZ::Lerp

### PR DESCRIPTION
**Pull request by pollend**
I replaced the usage of MCore::LinearInterpolate with AZ::Lerp. Any clue if there is a plan to move all the math utilities into Math and how would it be organized? 

MathUtils seems too general I guess would it make sense to split things across different headers and just have MathUtils collect those other referenced headers.

### CODE REVIEW

**Review by lsemp3d — APPROVED** [review]

File: Code/Framework/AzCore/AzCore/Math/MathUtils.h [hunk 1]
```diff
@@ -363,6 +363,13 @@ namespace AZ
         return a > b ? a : b;
     }
 
+    //! Returns a linear interpolation between 2 values.
+    template<typename T>
+    constexpr T Lerp(const T& a, const T& b, float t)
```
Comment (pollend) [inline Code/Framework/AzCore/AzCore/Math/MathUtils.h:368]: not sure I really like this though when there is already an implementation under Vector?

**Review by cgalvan — APPROVED** [review]
Did you replace all instances of `MCore::LinearInterpolate`?  If so, we should probably remove it as well so that someone doesn't re-introduce it.

**Comment by pollend** [comment]
> Did you replace all instances of `MCore::LinearInterpolate`?  If so, we should probably remove it as well so that someone doesn't re-introduce it.

Maybe just deprecate it for now. There are some other lerp functions in that algorithm file. How do we want to move that into math. Would it all be in mathutil?

File: Code/Framework/AzCore/AzCore/Math/MathUtils.h (hunk 1 above)
Comment (hultonha) [reply Code/Framework/AzCore/AzCore/Math/MathUtils.h:368]: Hmm so the `Vector3` lerp is optimized to use SIMD operations, internally it looks like this...

```c++
AZ_MATH_INLINE Vector3 Vector3::Lerp(const Vector3& dest, float t) const
{
    return Vector3(Simd::Vec3::Madd(Simd::Vec3::Sub(dest.m_value, m_value), Simd::Vec3::Splat(t), m_value));
}
```

So keeping both might well make sense (you could always profile each of them to compare 😛)

Comment (hultonha) [reply Code/Framework/AzCore/AzCore/Math/MathUtils.h:368]: Also I think due to overload resolution rules this template will actually be used for all cases now, so the `float`/`double` implementations are technically redundant (might be worth checking up on this). I think someone from sig-core might be worth taking a look (thoughts @lumberyard-employee-dm?)

Comment (pollend) [reply Code/Framework/AzCore/AzCore/Math/MathUtils.h:368]: we can get rid of the template and do something with constexpr to correctly handle those cases maaybe?

Comment (hultonha) [reply Code/Framework/AzCore/AzCore/Math/MathUtils.h:368]: I've added a few more reviewers to take a look, I think it does make sense to remove `MCore::LinearInterpolate` (if @amzn-jillich also agrees) but want to make sure others are okay with the new `Lerp` template. I'm fine with it in theory but I know in the recent past there has been discussion about avoiding templates for these sorts of things and I want to make sure other people get to weigh-in with any feedback/reasons why not to do it this way. If it's possible to just replace the usages of `MCore::LinearInterpolate` without adding the new template for now that might be the easiest option?

Comment (amzn-jillich) [reply Code/Framework/AzCore/AzCore/Math/MathUtils.h:368]: Is this actually needed? You've ported the vec3/quat occurences to the `source.Lerp(target, weight)` versions already. Or are there many more?

**Review by amzn-jillich — COMMENTED** [review]
Great idea and effort!

Did you do some testing on the changed areas as well?

**Comment by pollend** [comment]
> Great idea and effort!
> 
> Did you do some testing on the changed areas as well?

there is one templated usage in: [KeyTrackLinearDynamic.inl](https://github.com/o3de/o3de/pull/7809/files#diff-5ebdfd16c7f29e2cbe94e2e0d4aa9f5a7b213963b0dfa2f8a39ee38910f5b26b)

File: Code/Framework/AzCore/AzCore/Math/MathUtils.h (hunk 1 above)
Comment (hultonha) [reply Code/Framework/AzCore/AzCore/Math/MathUtils.h:368]: Awesome thanks @lumberyard-employee-dm, I think the template is then safe to add if we do want it. I remember there being something weird with templates around constructors being the 'best match' that could lead to some surprising behavior but good to know the overloads are selected first in this case 👍

**Review by hultonha — APPROVED** [review]
LGTM, I'm going to kick-off a build

**Comment by hultonha** [comment]
@pollend it looks like there are a few failures here - https://jenkins.build.o3de.org/job/O3DE/job/PR-7809/2/

**Comment by hultonha** [comment]
Hey @pollend, just wanted to check if you'd had time to take a look back at this PR and fix the build errors that popped up? Any update or know when you might get back to this? (If you're super busy we can move this to a draft for the time being). Thanks!

**Comment by pollend** [comment]
Try to fix it on the weekend.

**Comment by hultonha** [comment]
Thanks for updating @pollend! I've just kicked off another build here - https://jenkins.build.o3de.org/job/O3DE/view/change-requests/job/PR-7809/3/ Fingers crossed we get a green run this time, thanks! 🙂

**Comment by hultonha** [comment]
Merged! 🥳 Awesome work @pollend 🙂

**Comment by pollend** [comment]
how do we want to plan out migrating all those other lerp methods. do I place them all in utility or do we split it into a separate header. @hultonha

**Comment by hultonha** [comment]
Hmm it might be worth starting a discussion here - https://github.com/o3de/o3de/discussions and adding sig/core and feature/math if you can. I can tag other people who might be interested. Worth posting a link to the discussion on Discord too so others can see it. There we can probably hash out the best path forward.

**Pull request merged**